oaknational/Oak-Web-Application
# [CUR-941] Fix: filtered subject phase options 'gcse' if examboards present

## Description

Currently, only 'Core' units are visible for subjects with no valid examboard variants (e.g. Citizenship)
and 'GCSE' is not a valid KS4 option when there are exam board choices, it should be 'Core' and the exam board choices unless there is no exam board variance in that subject.

- GCSE pathways (e.g. Computer Science) are showing Core units, which they shoudln't
- RE (which has an empty core programme) is not displaying any KS4 units in the visualiser

Music year: 1956

- List of changes
- Filtered subject phase options to remove 'GCSE' ks4 option if examboard variants are present
- Physical education and computing core and gcse units should be separate - e.g. no core units present in the examboard / gcse variants

## How to test

1. Go to https://deploy-preview-2847--oak-web-application.netlify.thenational.academy/teachers/curriculum
2. Click on lot picker
3. You should now see the accurate KS4 options when Cycle 2 is enabled 

## Screenshots

How it used to look (delete if n/a):
{screenshots}
<img width="831" alt="Screenshot 2024-10-03 at 15 59 12" src="https://github.com/user-attachments/assets/ef922552-2286-4349-9c5d-9d09c406d64f">

<img width="842" alt="Screenshot 2024-10-03 at 15 59 39" src="https://github.com/user-attachments/assets/a5a42511-a3ef-44a0-a8c4-744740a925a5">

How it should now look:
<img width="844" alt="Screenshot 2024-10-03 at 15 57 51" src="https://github.com/user-attachments/assets/6ac21f92-9404-4bcd-a0da-2d122754c93a">

<img width="844" alt="Screenshot 2024-10-03 at 15 58 13" src="https://github.com/user-attachments/assets/18084710-1dde-4eb6-a1d4-b624e2d2df0d">
## Checklist
